kv/kvserver: TestReplicateQueueRebalance failed

kv/kvserver.TestReplicateQueueRebalance [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12670701?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12670701?buildTab=artifacts#/) on release-22.2 @ [579da0fa73e0d8427ab257a670b7d0d3f18c3701](https://github.com/cockroachdb/cockroach/commits/579da0fa73e0d8427ab257a670b7d0d3f18c3701):


```
=== RUN   TestReplicateQueueRebalance
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueRebalance2928933499
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
    replicate_queue_test.go:105: split off r53:/{Table/100-Max} [(n1,s1):1, (n2,s2):2, (n5,s5):3, (n3,s3):4, (n4,s4):5, next=6, gen=9, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:91: condition failed to evaluate within 45s: still downreplicating: r53:/{Table/100-Max} [(n1,s1):1, (n4,s4):5, (n5,s5):3, (n3,s3):4, next=6, gen=12, sticky=9223372036.854775807,2147483647]
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueRebalance2928933499
--- FAIL: TestReplicateQueueRebalance (1687.03s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalance.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>



Jira issue: CRDB-33501

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -68,6 +68,7 @@ func TestReplicateQueueRebalance(t *testing.T) {
 	// This test was seen taking north of 20m under race.
 	skip.UnderRace(t)
 	skip.UnderShort(t)
+	skip.UnderDeadlock(t)
 
 	const numNodes = 5
 

